compiler/src/dmd/expressionsem.d

```diff
@@ -13790,3 +13796,23 @@ Expression toBoolean(Expression exp, Scope* sc)
             return e;
     }
 }
+
+/* Push the necessary scope to solve `$` in `e1.ptr[$]` or `e1.ptr[0 .. $]` */
+private Scope* tryPushScopeForPtrDollar(Expression e1, Scope* sc)
+{
+    if (auto ce = e1.isCastExp())
+    {
+        if (auto tp = ce.type.isTypePointer())
+        {
+            const cty = ce.e1.type.ty;
+            const cop = ce.e1.op;
```

Comment:
why `cop` and `cty` ?
because of the const?
generally we don't use Hungarian notation.

Reply:
cty -> cast type
cop -> cast inner exp operator

TBH I just use old Walter's style for consistency.